diff --git a/node_modules/auth-astro/client.ts b/node_modules/auth-astro/client.ts
index 8a30c9e..8d3d340 100644
--- a/node_modules/auth-astro/client.ts
+++ b/node_modules/auth-astro/client.ts
@@ -44,7 +44,7 @@ export async function signIn<P extends RedirectableProviderType | undefined = un
 	const _signInUrl = `${signInUrl}?${new URLSearchParams(authorizationParams)}`
 
 	// TODO: Handle custom base path
-	const csrfTokenResponse = await fetch(`${prefix}/csrf`)
+	const csrfTokenResponse = await fetch(`${prefix}/csrf?${new URLSearchParams({ callbackUrl })}`);
 	const { csrfToken } = await csrfTokenResponse.json()
 
 	const res = await fetch(_signInUrl, {
